addons/stock/tests/test_stock_flow.py

```diff
@@ -1978,6 +1978,29 @@ def test_picking_scheduled_date_readonlyness(self):
 
         self.assertEqual(f.state, 'confirmed')
 
+    def test_stock_move_with_partner_id(self):
+        """ Ensure that the partner_id of the picking entry is
+        transmitted to the SM upon object creation.
+        """
+        partner_1 = self.env['res.partner'].create({'name': 'Hubert Bonisseur de la Bath'})
+        partner_2 = self.env['res.partner'].create({'name': 'Donald Clairvoyant du Bled'})
+        product = self.env['product.product'].create({'name': 'Un petit coup de polish', 'type': 'product'})
+        wh = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+
+        f = Form(self.env['stock.picking'])
+        f.partner_id = partner_1
+        f.picking_type_id = wh.out_type_id
+        with f.move_ids_without_package.new() as move:
+            move.product_id = product
+            move.product_uom_qty = 5
+        picking = f.save()
+
+        self.assertEqual(picking.move_lines.partner_id, partner_1)
+
+        f.partner_id = partner_2
```

Comment:
(I'm not sure you can reuse `f` like that, I think in this case you need to redefine it with `Form`, but I'm not sure)
Anyway, we don't want to use the form to test the writing phase, because it will trigger the onchange:
https://github.com/odoo/odoo/blob/f629c261cb7cd246f1f850388ce793420de5431a/addons/stock/models/stock_picking.py#L591-L598
And we know that this flow is already working properly
You should rather write on the picking directly
```suggestion
        picking.partner_id = partner02
```